subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveCachingStateStep.java

```diff
@@ -50,7 +50,7 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ResolveCachingStateStep.class);
     private static final CachingDisabledReason BUILD_CACHE_DISABLED_REASON = new CachingDisabledReason(CachingDisabledReasonCategory.BUILD_CACHE_DISABLED, "Build cache is disabled");
     private static final CachingState BUILD_CACHE_DISABLED_STATE = CachingState.disabledWithoutInputs(BUILD_CACHE_DISABLED_REASON);
-    private static final CachingDisabledReason VALIDATION_FAILED_REASON = new CachingDisabledReason(CachingDisabledReasonCategory.VALIDATION_FAILURE, "Validation failed");
+    private static final CachingDisabledReason VALIDATION_FAILED_REASON = new CachingDisabledReason(CachingDisabledReasonCategory.VALIDATION_FAILURE, "Caching has been disabled due to validation failures to ensure correctness. Please consult deprecation warnings for more details.");
```

Comment:
Suggestion:

Caching has been disabled to ensure correctness. Please consult deprecation warnings for more details.
As discussed, the term/concept of “validation failures” isn't meaningful to users an potentially confusing. I don't think it adds anything here and can be safely removed.